language/move-lang/README.md

```diff
@@ -17,15 +17,44 @@ Move source language is an ergonomic language for writing Modules and Scripts th
 * Move Build is used for checking and then compiling to bytecode
+
 In the future there should be other utilities for testing and play grounding the Move modules.
 
 There is unfortunately no documentation for the language syntax or features. See the stdlib for examples.
 
+## Design Principles
+
+### Mission
+
+Deliver a minimalistic, expressive, safe, and transparent language to produce--and link with--Move bytecode.
+
+### Primary Principles
+
+* **More Concise than Bytecode** Move is expression based, which lends itself to concise and composed programs without the need for extra locals or structure. In some sense, expressions allow for programming directly on the stack in a controlled environment, and in that way, the language gives the same level of functionality in a more concise and readable environment.
+
+* **Move Bytecode Transparency** The Move source language tries to lift up concepts in the Move bytecode into a source language; it is not trying to hide them. The bytecode already has some strong opinions (much stronger than you might expect to find in a bytecode language), and the source language is trying to keep that programming model and line of thinking. This means direct access to references, global storage, stack values, etc. The intention of this value is to remove the need to write bytecode directly. Additionally, this means full interoperability with functions and types declared in published modules.
+
+* **Stricter than Bytecode** The source language often adds additional levels of restrictions. At an expression level, this means no arbitrary manipulation of the stack (only can do so through expressions), and no dead code or unused effects. At a module level, this could mean additional warnings for unused types or un-invocable functions. At a conceptual/program level, this will also mean adding integration for formal verification.
+
+### Secondary Principles
+
+* **Pathway of Learning** Syntax choices and error messages are intended to give a natural flow of learning. For example, some of the choices around expression syntax could be made more familiar, but they would hurt the plug-n-play feeling of the expression based syntax, which might hurt developing a deeper understanding of the language.
+
+* **Aiding Common Community Patterns** As Move becomes more heavily used, common patterns for modules are likely to appear. Move might add new language features to make these patterns easier, clearer, or safer. But, they will not be added if it violates some other key design goal/value of the language.
+
+* **Careful Optimizations** Optimizations are an important developer tool, as they let a programmer write code in a more natural way. However, all of the optimizations performed must be semantic preserving, to prevent any catastrophic exploits or errors from occurring in optimized code. That being said, it is not the primary goal of the Move source language to produce *heavily* optimized code, but it is a nice feature to have.
```

Comment:
I would consider lifting up part of the second sentence as the principle: "Support semantic preserving optimization". I think this is much more concrete than "careful"